src/query/graphite/native/builtin_functions.go

```diff
@@ -556,11 +556,7 @@ func lowestCurrent(_ *common.Context, input singlePathSpec, n int) (ts.SeriesLis
 // windowSizeFunc calculates window size for moving average calculation
 type windowSizeFunc func(stepSize int) int
 
-// movingAverage calculates the moving average of a metric (or metrics) over a time interval.
-func movingAverage(ctx *common.Context, input singlePathSpec, windowSizeValue genericInterface) (*binaryContextShifter, error) {
-	if len(input.Values) == 0 {
-		return nil, nil
-	}
+func windowSizeParse(windowSizeValue genericInterface, input singlePathSpec) (time.Duration, windowSizeFunc, string, error) {
```

Comment:
also a nit: To make this read more literally, perhaps `func parseWindowSize(...)` might be a more readable function name?